server/src/main/java/com/linecorp/centraldogma/server/internal/storage/GarbageCollectingServicePlugin.java

```diff
@@ -0,0 +1,153 @@
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server.internal.storage;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import javax.annotation.Nullable;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Stopwatch;
+import com.google.common.util.concurrent.FutureCallback;
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableScheduledFuture;
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+
+import com.linecorp.armeria.common.util.TextFormatter;
+import com.linecorp.centraldogma.server.plugin.Plugin;
+import com.linecorp.centraldogma.server.plugin.PluginContext;
+import com.linecorp.centraldogma.server.plugin.PluginTarget;
+import com.linecorp.centraldogma.server.storage.project.Project;
+import com.linecorp.centraldogma.server.storage.project.ProjectManager;
+import com.linecorp.centraldogma.server.storage.repository.Repository;
+
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+public final class GarbageCollectingServicePlugin implements Plugin {
+
+    private static final Logger logger = LoggerFactory.getLogger(GarbageCollectingServicePlugin.class);
+
+    @Nullable
+    private ScheduledExecutorService gcWorker;
+    @Nullable
+    private ListenableScheduledFuture<?> scheduledFuture;
+
+    private volatile boolean stopping;
+
+    @Override
+    public PluginTarget target() {
+        return PluginTarget.ALL_REPLICAS;
+    }
+
+    @Override
+    public synchronized CompletionStage<Void> start(PluginContext context) {
+        requireNonNull(context, "context");
+
+        gcWorker = Executors.newSingleThreadScheduledExecutor(
+                new DefaultThreadFactory("repository-gc-worker", true));
+        final ListeningScheduledExecutorService scheduler = MoreExecutors.listeningDecorator(gcWorker);
+
+        // Run gc every day.
+        // TODO(ikhoon): Need to configure gc interval?
+        scheduledFuture = scheduler.scheduleAtFixedRate(() -> gc(context), 0, 1, TimeUnit.DAYS);
```

Comment:
> Shouldn't we schedule this when there's not so much usage instead of doing it right away after restarting?

Does that mean we need to start scheduling GC when the server traffic is low?

Reply:
Especially there's not so much read and write. 🤔

Reply:
Do we need to save the last revision at which GC was run? If so, the full GC will be executed only once when this code is deployed.

Reply:
I was sort of expecting to schedule at midnight so GC is not executed right after the instance is restarted which the instance is in a quite busy status.
But let's address @trustin's comment https://github.com/line/centraldogma/pull/564#issuecomment-790459669 first because we will probably need another way to schedule it.

Reply:
A Quartz scheduler expression was applied for more flexible gc scheduling. :-)